torch/csrc/jit/runtime/register_prim_ops_fulljit.cpp

```diff
@@ -621,6 +621,14 @@ RegisterOperators reg(
                    "Unsupported element type for tolist; only int, float and bool are supported");
              }
 
+             // Convert float Tensors to double since tolist is not supported
+             // for float Tensors.
+             if (t.scalar_type() == at::ScalarType::Float) {
+               TORCH_WARN(
+                   "Converting float Tensor to double because tolist is only supported for double type Tensors");
+               t = t.toType(at::ScalarType::Double);
```

Comment:
Seems this unconditionally casts FloatType to ScalarType. But if we did 
```
foo : List[int] = float_tensor.tolist()
```
we don't want to do this conversion or potentially warn the user about it. Seems like we should only perform casting if the user asks for `List[float]` and the tensors scalar type is `Double`.

Reply:
If the user asks for `List[float]` and the Tensor's scalar type is already `double`, what needs to be converted?

Reply:
Sorry I meant "tensor's scalar type is `float32`". My main point was that it's not necessary to do a cast from float32 to float64 if the user asked for `int` and we're going to error out on the next line (where we check `tryScalarTypeFromJitType(out_ty) == t.scalar_type()`).